### ISSUE TITLE
Create multiapi azure-core based client

### ISSUE THREAD

**Issue by iscai-msft** [Azure/azure-sdk-for-python#10179]
Right now, the python mutliapi SDK clients use the MultiApiClientMixin from azure.profiles.multiapiclient and it requires a dummy SDKClient on our part.

Create a multiapi azure-core based client that does not require this dummy SDKClient

**Comment by lmazuel** [comment]
Right now, when a package is “multi-api-ed”, in inherits from a mixin in azure-common:
https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/core/azure-common/azure/profiles/multiapiclient.py

We want to have a version of that in “azure-core”, so track2 don’t depend on “azure-common”.

Also, right now because of the way this mixing works, and some compatibility required, we have to generate a fake base class for multi api-client:
https://github.com/Azure/autorest.python/blob/06f4e34d6b47518001a217d5699195ea3cc49345/test/multiapi/Expected/AcceptanceTests/Multiapi/multiapi/_multiapi_service_client.py#L19-L24

TLDR This use is about:
-	Moving code from azure-common to azure-core in a way that makes us (and the architect 😉) happy
-	Remove code in the process that is tight to backward compatibility in the mixin
-	Make the hack in autorest with the fake base class not necessary.
